### PR TITLE
feat: added class for user error to prevent swallowing of errors to throw

### PR DESCRIPTION
**Pull request by jsalaber**
# Summary

Added a `UserError` class to distinguish between errors to be swallowed and errors to be thrown back to the user

### CODE REVIEW

File: sdk/nodejs/src/client.ts [hunk 1]
```diff
@@ -50,6 +51,9 @@ export class DVCClient {
         }
 
         const initializePromise = importBucketingLib({ options, logger: this.logger })
+            .catch((bucketingErr) => {
+                throw new UserError(`Error importing bucketing library: ${bucketingErr}`)
```
Comment (ajwootto) [inline sdk/nodejs/src/client.ts:55]: does this obscure the stack trace of the original error?

Comment (jsalaber) [reply sdk/nodejs/src/client.ts:55]: added support for creating it from `Error` objects

Comment (ajwootto) [reply sdk/nodejs/src/client.ts:55]: you'd have to pass the error in here then

File: sdk/nodejs/src/utils/userError.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export class UserError extends Error {
+    constructor(error: Error | string) {
+        super(error instanceof Error ? error.message : error)
+        this.name = 'UserError'
+        this.stack = error instanceof Error ? error.stack : undefined
```
Comment (ajwootto) [inline sdk/nodejs/src/utils/userError.ts:5]: does this make it so the stack isnt set properly when it's thrown with a string message?

Comment (jsalaber) [reply sdk/nodejs/src/utils/userError.ts:5]: this is for the cases where we throw an error when we know something is wrong and we throw with a string message

**Review by ajwootto — APPROVED** [review]

**Pull request merged**